### PR TITLE
Profile MVP Ready

### PR DESCRIPTION
**Pull request by manjilgtm**
mvp

### CODE REVIEW

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 1]
```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request,response);
+        //Checks whether user is logged in or not
+        String userName = (String) request.getSession().getAttribute("user");
```
Comment (tsalako) [inline src/main/java/codeu/controller/ProfileServlet.java:44]: Lets do some error checking on this userName string. See ChatServlet#doPost

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 2]
```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
...
+        System.out.println("The user store is " + userStore);
```
Comment (tsalako) [inline src/main/java/codeu/controller/ProfileServlet.java]: Please remove all the print lines before merging to master. Usually they are used for debugging so they should be cleaned up before a merge.

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 3]
```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
...
+        User loggedInUser = this.userStore.getUser(userName);
```
Comment (tsalako) [inline src/main/java/codeu/controller/ProfileServlet.java]: You shouldn't need the "this" reference.

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 4]
```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
...
+        System.out.println(loggedInUser.getDescription());
+        request.getSession().setAttribute("description", loggedInUser.getDescription());
+        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response)
             throws IOException, ServletException {
         String description = request.getParameter("description");
+        String userName = (String) request.getSession().getAttribute("user");
```
Comment (tsalako) [inline src/main/java/codeu/controller/ProfileServlet.java]: Error checking here as well

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 5]
```diff
@@ -37,13 +38,34 @@ void setUserStore(UserStore userStore) {
...
+        // read user from database using username
+        // get all the information of that loggedin user from datastore
+        User userToUpdate = this.userStore.getUser(userName);
+        //Setting Description of that Particular User in datastore
+        if (userToUpdate == null){
+            response.sendRedirect("/login");
+            return;
+        }
+        userToUpdate.setDescription(description);
```
Comment (tsalako) [inline src/main/java/codeu/controller/ProfileServlet.java]: When getting user input we want to be careful on html pages for code injection (https://en.wikipedia.org/wiki/Code_injection). We should use the Jsoup helper class to sanitize the description before we save it anywhere. Check out ChatServlet#doPost (you dont need to do the whole "processString" part)

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 6]
```diff
@@ -15,6 +15,10 @@
 <head>
     <title>Profile</title>
     <link rel="stylesheet" href="/css/main.css">
+    <!-- Latest compiled and minified CSS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
```
Comment (tsalako) [inline src/main/webapp/WEB-INF/view/profile.jsp:19]: what are we using this for currently?

File: src/test/java/codeu/controller/ProfileServletTest.java [hunk 7]
```diff
@@ -11,38 +12,71 @@
...
 
         profileServlet = new ProfileServlet();
         mockRequest = Mockito.mock(HttpServletRequest.class);
+        mockSession = Mockito.mock(HttpSession.class);
+
+        setupFakeUser();
+
+        mockUserStore = Mockito.mock(UserStore.class);
+        profileServlet.setUserStore(mockUserStore);
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
```
Comment (tsalako) [inline src/test/java/codeu/controller/ProfileServletTest.java]: Make sure you are doing your "Mockito.mock"s before your "Mockito.when"s or other functions that use these variables. This just helps make sure that the variable being tested isn't going to change between setup and execution.

File: src/test/java/codeu/controller/ProfileServletTest.java [hunk 8]
```diff
@@ -11,38 +12,71 @@
...
     @Test
     public void testdoGet() throws IOException, ServletException {
+        Mockito.when(mockSession.getAttribute("user")).thenReturn(TEST_USERNAME);
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+        Mockito.when(mockUserStore.getUser(TEST_USERNAME)).thenReturn(fakeUser);
+        System.out.println("name" + mockUserStore.getUser(TEST_USERNAME));
+
         profileServlet.doGet(mockRequest, mockResponse);
         Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```
Comment (tsalako) [inline src/test/java/codeu/controller/ProfileServletTest.java]: We also want to verify in the doGet that we are setting the description param properly.

**Review by tsalako — CHANGES_REQUESTED** [review]

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 2 above)
Comment (manjilgtm) [reply src/main/java/codeu/controller/ProfileServlet.java]: Done

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 3 above)
Comment (manjilgtm) [reply src/main/java/codeu/controller/ProfileServlet.java]: Removed!

File: src/test/java/codeu/controller/ProfileServletTest.java (hunk 7 above)
Comment (manjilgtm) [reply src/test/java/codeu/controller/ProfileServletTest.java]: Done!

File: src/main/webapp/WEB-INF/view/profile.jsp (hunk 6 above)
Comment (manjilgtm) [reply src/main/webapp/WEB-INF/view/profile.jsp:19]: I was using this to add some design to profile page.

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 5 above)
Comment (manjilgtm) [reply src/main/java/codeu/controller/ProfileServlet.java]: Fixed it.

File: src/main/webapp/WEB-INF/view/profile.jsp (hunk 6 above)
Comment (tsalako) [reply src/main/webapp/WEB-INF/view/profile.jsp:19]: Right. I just mean where specifically? The submit button?

Comment (manjilgtm) [reply src/main/webapp/WEB-INF/view/profile.jsp:19]: In submit button, Description Field and this template has got different font as well. Originally, It was like
new_one

File: src/test/java/codeu/controller/ProfileServletTest.java [hunk 9]
```diff
@@ -11,38 +12,73 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
 
 public class ProfileServletTest {
```
Comment (tsalako) [inline src/test/java/codeu/controller/ProfileServletTest.java:18]: Lets add tests to check the sanitation of the string.

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 1 above)
Comment (manjilgtm) [reply src/main/java/codeu/controller/ProfileServlet.java:44]: By error checking you mean verifying whether user is logged in or not.

Comment (ksingarajah) [reply src/main/java/codeu/controller/ProfileServlet.java:44]: +1

File: src/main/java/codeu/model/data/User.java [hunk 10]
```diff
@@ -44,12 +44,11 @@ public User(UUID id, String name, String passwordHash, Instant creation, String
   }
 
   /**
-   *
    * @param id the ID of this User
+   * @param name
```
Comment (ksingarajah) [inline src/main/java/codeu/model/data/User.java]: Probably delete this so you don't have duplicates.

File: src/main/java/codeu/model/data/User.java [hunk 11]
```diff
@@ -88,6 +87,5 @@ public void setDescription(String Description){
   /** Allows to know if the user is an admin */
   public Boolean isAdmin() {
     return (name.equals("Ricardo")||name.equals("Manjil")|| name.equals("Kirielle"));
```
Comment (ksingarajah) [inline src/main/java/codeu/model/data/User.java]: Can you add Tofe as well?

File: src/main/webapp/WEB-INF/view/profile.jsp (hunk 6 above)
Comment (ksingarajah) [reply src/main/webapp/WEB-INF/view/profile.jsp:19]: Looks nice!

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 12]
```diff
@@ -25,18 +29,22 @@
...
     <!--About Section -->
-    <h3>About You!</h3>
+    <h3>Summary</h3>
     <!-- info from descriptions -->
     <%=request.getSession().getAttribute("description")%>
     <br>
     <br>
     <br>
-    <h3> Edit your About Me (only you can see this)</h3>
+    <h3> Edit Summery</h3>
```
Comment (ksingarajah) [inline src/main/webapp/WEB-INF/view/profile.jsp]: Summary*

File: src/test/java/codeu/controller/ProfileServletTest.java (hunk 9 above)
Comment (ksingarajah) [reply src/test/java/codeu/controller/ProfileServletTest.java:18]: +1

**Review by ksingarajah — CHANGES_REQUESTED** [review]
Just a few changes for typos/duplicates, and added testing for sanitation.

File: src/test/java/codeu/controller/ProfileServletTest.java (hunk 9 above)
Comment (manjilgtm) [reply src/test/java/codeu/controller/ProfileServletTest.java:18]: Done!

File: src/test/java/codeu/controller/ProfileServletTest.java (hunk 8 above)
Comment (manjilgtm) [reply src/test/java/codeu/controller/ProfileServletTest.java]: Fixed!

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 1 above)
Comment (tsalako) [reply src/main/java/codeu/controller/ProfileServlet.java:44]: Nope. Verify that the string is not null

File: src/main/java/codeu/model/data/User.java (hunk 11 above)
Comment (tsalako) [reply src/main/java/codeu/model/data/User.java]: Ricardo has already changed this admin stuff you probably dont need to.

**Review by tsalako — COMMENTED** [review]
Do a pull from origin master to this branch. You may have some merge conflicts to resolve.

File: src/main/java/codeu/controller/ProfileServlet.java (hunk 1 above)
Comment (manjilgtm) [reply src/main/java/codeu/controller/ProfileServlet.java:44]: Fiixed it!

File: src/main/java/codeu/model/data/User.java (hunk 11 above)
Comment (manjilgtm) [reply src/main/java/codeu/model/data/User.java]: Got it!

**Review by tsalako — CHANGES_REQUESTED** [review]
Looks good but lets have one more commit for the merge of the current master to this branch.

**Review by ksingarajah — APPROVED** [review]

File: pom.xml [hunk 13]
```diff
@@ -106,6 +106,11 @@
       <version>1.9.60</version>
       <scope>test</scope>
     </dependency>
+      <dependency>
```
Comment (tsalako) [inline pom.xml]: What is this dependency for? I think its already in this file.

Comment (manjilgtm) [reply pom.xml]: My IDE suggested "Mavern Projects needed to be imported" When i tried running without importing it didn't run.

Comment (manjilgtm) [reply pom.xml]: I also don't know why it asked for!!

Comment (tsalako) [reply pom.xml]: Can you try running this with it deleted again? Also if you get the error again can you share it?

Comment (manjilgtm) [reply pom.xml]: I copied whole pom.XML file from master and commit again in my branch and it works fine. I think i messed something there before. It should be fine now.

**Review by tsalako — APPROVED** [review]
LGTM

**Pull request merged**